clusterDeployer.py

```diff
@@ -447,26 +446,19 @@ def create_workers(self) -> None:
         for h in hosts_with_workers:
             h.ensure_images(iso_file, infra_env, masters=False)
 
-        futures = []
-        for h in hosts_with_workers:
-            futures.extend(h.start_workers(infra_env, executor))
+        # The _start_node function submits another Future, we need different executor to prevent deadlocks
+        # https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor
+        inner_executor = ThreadPoolExecutor(max_workers=len(self._cc.workers))
```

Comment:
Can you simplify this? Can you elaborate what you're trying to do here?

Reply:
The installation step needs to run in background so it needs different executor because we cannot spawn future from future.

Reply:
Can we spawn the future from the main thread?

Reply:
Sorry I wasn't probably clear enough, we cannot spawn future from future if they are on different executors.